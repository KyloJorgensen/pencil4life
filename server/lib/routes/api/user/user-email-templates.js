'use strict';
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var react_html_email_1 = require("react-html-email");
var variables_express_1 = require("../../../config/variables.express");
exports.resetPasswordEmailTemplate = function (params) { return react_html_email_1.renderEmail(React.createElement(react_html_email_1.Email, { title: "Pencil4life Password Reset" },
    React.createElement(react_html_email_1.Box, { width: '600px', cellPadding: 8, cellSpacing: 16, align: 'center' },
        React.createElement(react_html_email_1.Item, { align: "center" },
            React.createElement(react_html_email_1.A, { href: "https://" + (variables_express_1.NODE_ENV == 'development' ? 'dev' : 'www') + ".pencil4life.com/#/" },
                React.createElement(react_html_email_1.Image, { src: 'https://pencil4life.com/images/pencil4lifelogo-blue-transparent.png', width: 600, height: 155, alt: 'PENCIL4Life' }))),
        React.createElement(react_html_email_1.Item, null,
            React.createElement(react_html_email_1.Span, { fontSize: 18 },
                "Hello ",
                params.username,
                ",")),
        React.createElement(react_html_email_1.Item, null,
            React.createElement(react_html_email_1.Span, { fontSize: 12 }, "You recently requested to reset your password for your Pencil4Life Account. Use the button below to reset it."),
            React.createElement(react_html_email_1.Span, { fontSize: 12, fontWeight: 'bold' }, "This password reset is only vaild for the next 24 hours.")),
        React.createElement(react_html_email_1.Item, { align: "center" },
            React.createElement(react_html_email_1.Span, { fontSize: 12 },
                React.createElement(react_html_email_1.A, { href: "https://" + (variables_express_1.NODE_ENV == 'development' ? 'dev' : 'www') + ".pencil4life.com/#/resetpassword/" + params.userId + "/" + params.code }, "Click Here"),
                " to reset your password.")),
        React.createElement(react_html_email_1.Item, null,
            React.createElement(react_html_email_1.Span, { fontSize: 12 },
                "If you didn't make this request no need to worry, but if you have questions about your accounts security  ",
                React.createElement(react_html_email_1.A, { href: "https://" + (variables_express_1.NODE_ENV == 'development' ? 'dev' : 'www') + ".pencil4life.com/#/contact/message/" }, "Click Here"),
                " to send us a message.")),
        React.createElement(react_html_email_1.Item, null,
            React.createElement(react_html_email_1.Span, { fontSize: 12 }, "Thank you, Pencil 4 Life"))))); };
exports.invaildResetPasswordEmailTemplate = function () { return react_html_email_1.renderEmail(React.createElement(react_html_email_1.Email, { title: "Pencil4life Password Reset" },
    React.createElement(react_html_email_1.Box, { width: '600px', cellPadding: 8, cellSpacing: 16, align: 'center' },
        React.createElement(react_html_email_1.Item, { align: "center" },
            React.createElement(react_html_email_1.A, { href: "https://" + (variables_express_1.NODE_ENV == 'development' ? 'dev' : 'www') + ".pencil4life.com/#/" },
                React.createElement(react_html_email_1.Image, { src: 'https://pencil4life.com/images/pencil4lifelogo-blue-transparent.png', width: 600, height: 155, alt: 'PENCIL4Life' }))),
        React.createElement(react_html_email_1.Item, null,
            React.createElement(react_html_email_1.Span, { fontSize: 20 }, "Hello Friend,")),
        React.createElement(react_html_email_1.Item, null,
            React.createElement(react_html_email_1.Span, { fontSize: 12 }, "You recently requested to reset your password for your Pencil4Life Account. But we don't have any account using your email address.")),
        React.createElement(react_html_email_1.Item, { align: "center" },
            React.createElement(react_html_email_1.Span, { fontSize: 12 },
                React.createElement(react_html_email_1.A, { href: "https://" + (variables_express_1.NODE_ENV == 'development' ? 'dev' : 'www') + ".pencil4life.com/#/signup" }, "Click Here"),
                " to make an account.")),
        React.createElement(react_html_email_1.Item, null,
            React.createElement(react_html_email_1.Span, { fontSize: 12 },
                "If you didn't make this request no need to worry, but if you have questions about your accounts security  ",
                React.createElement(react_html_email_1.A, { href: "https://" + (variables_express_1.NODE_ENV == 'development' ? 'dev' : 'www') + ".pencil4life.com/#/contact/message/" }, "Click Here"),
                " to send us a message.")),
        React.createElement(react_html_email_1.Item, null,
            React.createElement(react_html_email_1.Span, { fontSize: 12 }, "Thank you, Pencil 4 Life"))))); };
