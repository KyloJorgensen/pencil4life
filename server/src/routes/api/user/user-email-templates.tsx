'use strict';

import * as React from 'react';
import { Email, Item, Span, Box, A, renderEmail, Image } from 'react-html-email';
import { NODE_ENV } from '../../../config/variables.express';

export interface ResetPasswordEmailTemplateParams {
    userId: string;
    username: string;
    code: string;
}

export const resetPasswordEmailTemplate = (params: ResetPasswordEmailTemplateParams) => renderEmail(
    <Email title="Pencil4life Password Reset">
        <Item>
            <Span fontSize={20}>
                Hello {params.username},
            </Span>
        </Item>
        <Item>
            <Span fontSize={12}>You recently requested to reset your password for your Pencil4Life Account. Use the button below to reset it.</Span>
            <Span fontWeight={'bold'}>This password reset is only vaild for the next 24 hours.</Span>
        </Item>
        <Item align="center">
            <Span fontSize={12}>
                <A href={`https://${NODE_ENV == 'development' ? 'dev' :'www'}.pencil4life.com/#/resetpassword/${params.userId}/${params.code}`}>Click Here</A> to reset your password.
            </Span>
        </Item>
    </Email>
);

export const invaildResetPasswordEmailTemplate = () => renderEmail(
    <Email title="Pencil4life Password Reset">
        <Item>
            <Span fontSize={20}>
                Hello Friend,
            </Span>
        </Item>
        <Item>
            <Span fontSize={12}>You recently requested to reset your password for your Pencil4Life Account. But we don't have any account using your email address.</Span>
        </Item>
        <Item align="center">
            <Span fontSize={12}>
                <A href={`https://${NODE_ENV == 'development' ? 'dev' :'www'}.pencil4life.com/#/signup`}>Click Here</A> to make an account.
            </Span>
        </Item>
    </Email>
);



// Documentation

// Components
// Components in react-html-email include defaults for basic style properties, so that client styles are reset and normalized. Every component accepts a style prop which overrides the reset styles.

// <Email>
// An HTML document with a centered 600px <table> inside <table> container based on HTML Email Boilerplate.

// It's necessary to always include a title prop for some clients' "open in browser" feature.

// See MailChimp's HTML guide for how this works.

// <Box>
// A simplification of the <table> element, the workhorse of an HTML email design. <Box>es contain a vertical stack of <Item>s. Use them to create visual structure, filled buttons, and spacing.

// <Item>
// A subsection of a <Box>, essentially a <tr><td> unit.

// <Span>
// Use to assign styles to text.

// It can be handy to create an object containing your default text styles for reuse. For example:

// const textStyles = {
//   fontFamily: 'Verdana',
//   fontSize: 42,
//   fontWeight: 'bold',
//   color: 'orange',
// }
 
// [...]
 
// <Span {...textDefaults}>Congratulations!</Span>
// <Span {...textDefaults}>You won a free cruise!</Span>
// <A>
// Use to format links. Requires an href prop. Always sets target="_blank" and defaults to underline. To remove the underline, set textDecoration="none".

// <Image>
// An image, without any pesky borders, outlines, or underlines by default. Requires a src prop, and width and height to be set. You can override the default styles (such as adding a border) using the style prop.

// Head CSS and Media Queries
// You can pass a string prop headCSS to your <Email> component. You can see it in our kitchenSink.js example.