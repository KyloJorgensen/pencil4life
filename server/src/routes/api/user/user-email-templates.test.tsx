'use strict';

import { resetPasswordEmailTemplate, invaildResetPasswordEmailTemplate } from "./user-email-templates";

describe('Reset Password Email Template', () => {
    it('should match snapshot', () => {
        expect(resetPasswordEmailTemplate({
            userId: 'user_id',
            username: 'username',
            code: 'reset_code'
        })).toMatchSnapshot();
    });
});

describe('Invaild Reset Password Email Template', () => {
    it('should match snapshot', () => {
        expect(invaildResetPasswordEmailTemplate()).toMatchSnapshot();
    });
});