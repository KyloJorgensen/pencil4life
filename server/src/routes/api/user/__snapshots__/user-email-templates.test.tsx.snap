// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Invaild Reset Password Email Template should match snapshot 1`] = `"<!DOCTYPE html PUBLIC \\"-//W3C//DTD XHTML 1.0 Strict//EN\\" \\"http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd\\"><html lang=\\"en\\" xmlns=\\"http://www.w3.org/1999/xhtml\\"><head><meta http-equiv=\\"Content-Type\\" content=\\"text/html; charset=utf-8\\"/><meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\"/><title>Pencil4life Password Reset</title></head><body style=\\"width:100%;margin:0;padding:0;-webkit-text-size-adjust:100%;-ms-text-size-adjust:100%\\"><table width=\\"100%\\" height=\\"100%\\" cellPadding=\\"0\\" cellSpacing=\\"0\\" border=\\"0\\" align=\\"left\\" valign=\\"top\\"><tbody><tr><td align=\\"center\\" valign=\\"top\\"><table width=\\"600\\" align=\\"center\\" cellPadding=\\"0\\" cellSpacing=\\"0\\" border=\\"0\\" valign=\\"top\\"><tbody><table width=\\"600px\\" cellPadding=\\"8\\" cellSpacing=\\"16\\" align=\\"center\\" border=\\"0\\" valign=\\"top\\"><tbody><tr><td align=\\"center\\"><a href=\\"https://www.pencil4life.com/#/\\" target=\\"_blank\\" style=\\"text-decoration:underline\\"><img alt=\\"PENCIL4Life\\" src=\\"https://pencil4life.com/images/pencil4lifelogo-blue-transparent.png\\" width=\\"600\\" height=\\"155\\" style=\\"display:block;outline:none;border:none;text-decoration:none\\"/></a></td></tr><tr><td><span style=\\"font-family:sans-serif;font-size:20px;line-height:20px;color:#000\\">Hello Friend,</span></td></tr><tr><td><span style=\\"font-family:sans-serif;font-size:12px;line-height:12px;color:#000\\">You recently requested to reset your password for your Pencil4Life Account. But we don&#x27;t have any account using your email address.</span></td></tr><tr><td align=\\"center\\"><span style=\\"font-family:sans-serif;font-size:12px;line-height:12px;color:#000\\"><a href=\\"https://www.pencil4life.com/#/signup\\" target=\\"_blank\\" style=\\"text-decoration:underline\\">Click Here</a> to make an account.</span></td></tr><tr><td><span style=\\"font-family:sans-serif;font-size:12px;line-height:12px;color:#000\\">If you didn&#x27;t make this request no need to worry, but if you have questions about your accounts security  <a href=\\"https://www.pencil4life.com/#/contact/message/\\" target=\\"_blank\\" style=\\"text-decoration:underline\\">Click Here</a> to send us a message.</span></td></tr><tr><td><span style=\\"font-family:sans-serif;font-size:12px;line-height:12px;color:#000\\">Thank you, Pencil 4 Life</span></td></tr></tbody></table></tbody></table></td></tr></tbody></table></body></html>"`;

exports[`Reset Password Email Template should match snapshot 1`] = `"<!DOCTYPE html PUBLIC \\"-//W3C//DTD XHTML 1.0 Strict//EN\\" \\"http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd\\"><html lang=\\"en\\" xmlns=\\"http://www.w3.org/1999/xhtml\\"><head><meta http-equiv=\\"Content-Type\\" content=\\"text/html; charset=utf-8\\"/><meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\"/><title>Pencil4life Password Reset</title></head><body style=\\"width:100%;margin:0;padding:0;-webkit-text-size-adjust:100%;-ms-text-size-adjust:100%\\"><table width=\\"100%\\" height=\\"100%\\" cellPadding=\\"0\\" cellSpacing=\\"0\\" border=\\"0\\" align=\\"left\\" valign=\\"top\\"><tbody><tr><td align=\\"center\\" valign=\\"top\\"><table width=\\"600\\" align=\\"center\\" cellPadding=\\"0\\" cellSpacing=\\"0\\" border=\\"0\\" valign=\\"top\\"><tbody><table width=\\"600px\\" cellPadding=\\"8\\" cellSpacing=\\"16\\" align=\\"center\\" border=\\"0\\" valign=\\"top\\"><tbody><tr><td align=\\"center\\"><a href=\\"https://www.pencil4life.com/#/\\" target=\\"_blank\\" style=\\"text-decoration:underline\\"><img alt=\\"PENCIL4Life\\" src=\\"https://pencil4life.com/images/pencil4lifelogo-blue-transparent.png\\" width=\\"600\\" height=\\"155\\" style=\\"display:block;outline:none;border:none;text-decoration:none\\"/></a></td></tr><tr><td><span style=\\"font-family:sans-serif;font-size:18px;line-height:18px;color:#000\\">Hello username,</span></td></tr><tr><td><span style=\\"font-family:sans-serif;font-size:12px;line-height:12px;color:#000\\">You recently requested to reset your password for your Pencil4Life Account. Use the button below to reset it.</span><span style=\\"font-family:sans-serif;font-size:12px;font-weight:bold;line-height:12px;color:#000\\">This password reset is only vaild for the next 24 hours.</span></td></tr><tr><td align=\\"center\\"><span style=\\"font-family:sans-serif;font-size:12px;line-height:12px;color:#000\\"><a href=\\"https://www.pencil4life.com/#/resetpassword/user_id/reset_code\\" target=\\"_blank\\" style=\\"text-decoration:underline\\">Click Here</a> to reset your password.</span></td></tr><tr><td><span style=\\"font-family:sans-serif;font-size:12px;line-height:12px;color:#000\\">If you didn&#x27;t make this request no need to worry, but if you have questions about your accounts security  <a href=\\"https://www.pencil4life.com/#/contact/message/\\" target=\\"_blank\\" style=\\"text-decoration:underline\\">Click Here</a> to send us a message.</span></td></tr><tr><td><span style=\\"font-family:sans-serif;font-size:12px;line-height:12px;color:#000\\">Thank you, Pencil 4 Life</span></td></tr></tbody></table></tbody></table></td></tr></tbody></table></body></html>"`;
